db/version_edit.h

```diff
@@ -463,22 +464,27 @@ class VersionEdit {
   bool IsWalAddition() const { return !wal_additions_.empty(); }
 
   // Delete a WAL (either directly deleted or archived).
-  void DeleteWal(WalNumber number) {
-    assert(NumEntries() == wal_deletions_.size());
-    wal_deletions_.emplace_back(number);
+  // AddWal and DeleteWalsBefore cannot be called on the same VersionEdit.
+  void DeleteWalsBefore(WalNumber number) {
+    assert(NumEntries() == !wal_deletion_.IsEmpty());
```

Comment:
How about `assert((NumEntries() > 0) == !wal_deletion_.IsEmpty());` to make the intention clearer?